src/models/searchservice/request/StaticFilter.ts

```diff
@@ -5,9 +5,49 @@ import { FieldValueFilter } from './FieldValueFilter';
  *
  * @public
  */
-export type StaticFilter =
-  { kind: 'fieldValue', value: FieldValueFilter } |
-  { kind: 'combination', combinator: FilterCombinator, children: StaticFilter[] };
+export type StaticFilter = FieldValueStaticFilter | DisjunctionStaticFilter | ConjunctionStaticFilter;
+
+/**
+ * A {@link FieldValueFilter} with the kind of filter specified
+ * to discriminate between static filter types.
+ *
+ * @public
+ */
+export interface FieldValueStaticFilter extends FieldValueFilter {
+  /** The kind of static filter. */
+  kind: 'fieldValue'
```

Comment:
Have we thought about using an enum for filter kind? Not saying we should or shouldn't, I like the simplicity just making it a string is but we also seem to prefer enums in this library

Reply:
I didn't really see a need because we don't currently refer to the `kind` values collectively. I think an enum makes sense in cases like for `Matcher` where we do refer to all the values as a single type that users can choose between